include/fiction/algorithms/physical_design/design_sidb_gates.hpp

```diff
@@ -0,0 +1,374 @@
+//
+// Created by Jan Drewniok on 11.09.23.
+//
+
+#ifndef FICTION_DESIGN_SIDB_GATES_HPP
+#define FICTION_DESIGN_SIDB_GATES_HPP
+
+#include "fiction/algorithms/simulation/sidb/is_operational.hpp"
+#include "fiction/algorithms/simulation/sidb/operational_domain.hpp"
+#include "fiction/algorithms/simulation/sidb/random_sidb_layout_generator.hpp"
+#include "fiction/algorithms/simulation/sidb/sidb_simulation_engine.hpp"
+#include "fiction/algorithms/simulation/sidb/sidb_simulation_parameters.hpp"
+#include "fiction/layouts/coordinates.hpp"
+#include "fiction/traits.hpp"
+#include "fiction/types.hpp"
+#include "fiction/utils/layout_utils.hpp"
+#include "fiction/utils/math_utils.hpp"
+#include "fiction/utils/truth_table_utils.hpp"
+
+#include <kitty/dynamic_truth_table.hpp>
+
+#include <algorithm>
+#include <atomic>
+#include <cassert>
+#include <cstdint>
+#include <cstdlib>
+#include <future>
+#include <numeric>
+#include <thread>
+#include <utility>
+#include <vector>
+
+#include <combinations.h>
+
+namespace fiction
+{
+
+/**
+ * This struct contains parameters and settings to design SiDB gates.
+ *
+ */
+struct design_sidb_gates_params
+{
+    /**
+     * Selector for the available design approaches.
+     */
+    enum class design_sidb_gates_mode
+    {
+        /**
+         * All (exhaustive) gate layouts are designed.
+         */
+        EXHAUSTIVE,
+        /**
+         * Gate layouts are designed randomly.
+         */
+        RANDOM
+    };
+    /**
+     * Selected design mode.
+     */
+    design_sidb_gates_mode design_mode = design_sidb_gates_mode::EXHAUSTIVE;
+    /**
+     * All Parameters for physical SiDB simulations.
+     */
+    sidb_simulation_parameters phys_params{};
```

Comment:
In all other algorithms that involve simulation, we have the `sidb_simulation_parameters` attribute first in the order of parameters. Since this is a `struct' with an automatically defined default constructor, the order of elements here matters.

Reply:
@marcelwa oh, thank you!